WordPress/Classes/Models/BasePost.m

```diff
@@ -26,45 +34,77 @@ @implementation BasePost
+        return NSLocalizedString(@"Published", @"Name for the status of a published post.");
+
+    } else if ([status isEqualToString:PostStatusTrash]) {
+        return NSLocalizedString(@"Trashed", @"Name for the status of a trashed post");
+
+    } else if ([status isEqualToString:PostStatusScheduled]) {
+        return NSLocalizedString(@"Scheduled", @"Name for the status of a scheduled post");
     }
 
     return status;
 }
 
-+ (NSString *)statusForTitle:(NSString *)title
++ (NSString *)makePlainText:(NSString *)string
 {
-    if ([title isEqualToString:NSLocalizedString(@"Draft", @"")]) {
-        return @"draft";
-    } else if ([title isEqualToString:NSLocalizedString(@"Pending review", @"")]) {
-        return @"pending";
-    } else if ([title isEqualToString:NSLocalizedString(@"Private", @"")]) {
-        return @"private";
-    } else if ([title isEqualToString:NSLocalizedString(@"Published", @"")]) {
-        return @"publish";
-    }
+    NSCharacterSet *charSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
+    return [[[string stringByStrippingHTML] stringByDecodingXMLCharacters] stringByTrimmingCharactersInSet:charSet];
+}
+
++ (NSString *)createSummaryFromContent:(NSString *)string
+{
+    string = [self makePlainText:string];
+    string = [self stripShortcodesFromString:string];
+    string = [string stringByTrimmingCharactersInSet:[NSCharacterSet characterSetWithCharactersInString:@"\n"]];
+    return [string stringByEllipsizingWithMaxLength:PostDerivedSummaryLength preserveWords:YES];
+}
 
-    return title;
++ (NSString *)stripShortcodesFromString:(NSString *)string
+{
+    static NSRegularExpression *regex;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        NSError *error;
+        NSString *pattern = @"\\[[^\\]]+\\]";
+        regex = [[NSRegularExpression alloc] initWithPattern:pattern options:NSRegularExpressionCaseInsensitive error:&error];
+        if (error) {
+            DDLogError(@"Error parsing regex: %@", error);
+        }
+    });
+    NSRange range = NSMakeRange(0, [string length]);
+    return [regex stringByReplacingMatchesInString:string
+                                           options:NSMatchingReportCompletion
+                                             range:range
+                                      withTemplate:@""];
 }
 
-- (NSArray *)availableStatuses
+- (NSArray *)availableStatusesForEditing
 {
-    return @[NSLocalizedString(@"Draft", @""),
-             NSLocalizedString(@"Pending review", @""),
-             NSLocalizedString(@"Private", @""),
-             NSLocalizedString(@"Published", @"")];
+    // Subset of status a user may assign to a post they are editing.
+    // Private is not listed as this is determined by the visibility settings.
+    // Scheduled is not listed as this should be handled by assigning a
+    // future date.
+    // Trash is not listed as this should be handled via a delete action.
```

Comment:
I'd suggest moving this comment to the header file in the `@details` section, as it adds a bit more insight into what this method does.  I was going to mention it's purpose wasn't so clear from the header file comment - this info could be a good addition there.